### ISSUE TITLE
Keypoints normalization suggestion

### ISSUE THREAD

**Issue by Golbstein** [vishalsharbidar/SphereGlue#8]
Since you're working on a spherical images, the keypoints become cyclical. Normalizing xy between -1 and 1 doesn't make much sense.

You can convert the spherical pixels to cartesian points (on a unit sphere) and use this as your normalization

```
def sph2cart(spherical):
    return torch.stack([torch.sin(spherical[..., 0]) * torch.cos(spherical[..., 1]),
                        torch.sin(spherical[..., 0]) * torch.sin(spherical[..., 1]),
                        torch.cos(spherical[..., 0])], dim=-1)

def pix2sph(kp, h):
    phi = torch.pi * (1 - kp[..., 0]/h)
    theta = torch.pi * kp[..., 1] / h
    return torch.stack((theta, phi), dim=-1)


def normalize_keypoints(kpts, shape_or_size):
    h, w = shape_or_size[-2:]
    return sph2cart(pix2sph(kpts, h))

```
